src/tests/lossy_link.cpp

```diff
@@ -0,0 +1,10 @@
+#include "lossy_link.h"
+
+bool LossyLink::drop(float ratio)
+{
+    std::lock_guard<std::mutex> lock(_mutex);
+
+    std::uniform_real_distribution<> distribution(0.0f, 1.0f);
+    const float random_value = distribution(_engine);
```

Comment:
I thought about this some more and I think it's better to use a separate engine (or `LossyLink` instance) for RX and TX, and init with a different seed.
The problem that can happen now is: whether a message is dropped or not depends on ordering of incoming vs outgoing messages. For example a random incoming message (e.g. heartbeat) might arrive just before we send a message, or it arrives directly after.
So there is timing-dependent behavior. Separating RX from TX will help making tests more reproducible.

We can increase reproducibility even more if we apply `drop` to only a specific set of incoming mavlink messages (e.g. all mission messages), so that a test is for example independent from streamed attitude messages.

Reply:
Right, good point. I'll change it.

Reply:
Done in f127017.